### PR TITLE
Add possibility to define routes

### PR DESCRIPTION
**Pull request by sebastianrakel**
Add route configuration possibility to this module

### CODE REVIEW

File: manifests/interface.pp [hunk 1]
```diff
@@ -156,7 +158,12 @@
   }
 
   systemd::network { "${interface}.network":
-    content         => epp("${module_name}/network.epp", { 'interface' => $interface, 'addresses' => $addresses }),
+    content         => epp("${module_name}/network.epp",
+                           {
+                             'interface' => $interface,
+                             'addresses' => $addresses,
+                             'routes'    => $routes,
+                           }),
```
Comment (root-expert) [inline manifests/interface.pp:166]: Try to add a newline between } and ). 
Probably that will fix rubocop error?

File: manifests/interface.pp [hunk 2]
```diff
@@ -87,6 +88,7 @@
   Optional[String[1]] $description = undef,
   Optional[Integer[1280, 9000]] $mtu = undef,
   Optional[String[1]] $public_key = undef,
+  Optional[Array[Hash]] $routes = undef,
```
[the remark this answers is not in the document]
Comment (bastelfreak) [reply manifests/interface.pp:91]: We can probably make the type more strict? Like `Array[Hash[String[1],String[1]]]`

File: templates/network.epp [hunk 3]
```diff
@@ -22,4 +23,10 @@ KeepConfiguration=yes
 <%= $key %>=<%= $value %>
 <% } -%>
 <% } -%>
+<% $routes.each |$route| { -%>
 
+[Route]
+<% $route.each |$key, $value| { -%>
+<%= $key %>=<%= $value %>
+<% } -%>
+<% } -%>
```
Comment (bastelfreak) [inline templates/network.epp:32]: should we add a newline between line 31 and 32? that ensures that the file terminates properly with \n and the `[Route]` blocks will have a newline between

Comment (bastelfreak) [reply templates/network.epp:32]: I think this needs to be:
```suggestion
<% } %>
```

to have a proper line ending

Comment (sebastianrakel) [reply templates/network.epp:32]: its copied from above so the file ending keeps the same.
on my system after rollout there is a proper file ending

**Pull request merged**